lib/arclight/repository.rb

```diff
@@ -59,11 +59,18 @@ def self.all
       from_yaml(ENV['REPOSITORY_FILE'] || 'config/repositories.yml').values
     end
 
-    # Mimics ActiveRecord `find_by` behavior
+    # Mimics ActiveRecord dynamic `find_by` behavior for the slug or name
     #
-    # @param [String] `slug`
-    def self.find_by(slug)
-      all.select { |repo| repo.slug == slug }.first
+    # @param [String] `slug` or `name`
+    # @return [Repository]
+    def self.find_by(slug: nil, name: nil)
+      if slug
+        all.select { |repo| repo.slug == slug }.first
```

Comment:
I mean using `all.find { |repo| repo.slug == slug }` instead of `all.select { |repo| repo.slug == slug }.first`

`select` iterates through the entire array that you call it on, and `find` will find the first and exit, which would make sense since we're simply calling `first` anyway.